templates/test/test/runtests.jl

```diff
@@ -1,12 +1,36 @@
-@eval module $(gensym())
-using Test: @testset
+using SafeTestsets
+using Suppressor: @suppress
 
-@testset "{PKGNAME}.jl" begin
-  filenames = filter(readdir(@__DIR__)) do f
-    startswith("test_")(f) && endswith(".jl")(f)
+# check for filtered groups
+# either via `--group=ALL` or through ENV["GROUP"]
+const pat = r"(?:--group=)(\w+)"
+arg_id = findfirst(contains(pat), ARGS)
+const GROUP = uppercase(
+  if isnothing(arg_id)
+    get(ENV, "GROUP", "ALL")
+  else
+    only(match(pat, ARGS[arg_id]).captures)
+  end,
+)
+
+@time begin
```

Comment:
Doesn't `@testset` report times now?

Reply:
I.e. could we achieve this by wrapping all of this in `@testset` or `@safetestset`?

Reply:
I'm not entirely sure, I mostly copied this from my other workflows. I think the main difference is that `@time` also reports splits in compiletime/runtime, which I find interesting, but I have absolutely no preference about this.
Just as an aside, wrapping things in nested `@testset` structures makes it such that output is only printed after all tests finish. Not a big deal, but something to keep in mind.